test/removeFromTeam.js

```diff
@@ -47,6 +79,56 @@ describe("removing a user from a team", function() {
         );
     }));
 
-    // TODO: add tests for more hidden groups other than just in announcements
+    it("should remove the user from event hidden groups", coroutine(function*() {
+        let date = new Date();
+        let events = yield sessions[0]("GET",
+            `/events/startYear/${date.getFullYear() - 1}/startMonth/0
+                /endYear/${date.getFullYear() + 1}/endMonth/0`);
+        assert.equal(events[0].audience.users.length, 1,
+            "a user was removed from event hidden group"
+        );
+        assert.equal(events[0].audience.users[0], data.users[0]._id,
+            "the correct user was removed from event hidden group"
+        );
+    }));
+
+    it("should remove the user from group chat hidden groups", coroutine(function*() {
+        let chats = yield sessions[0]("GET", "/chats");
+        let groupChat = chats.find(chat => !chat.isTwoPeople);
+        assert.equal(groupChat.audience.users.length, 1,
+            "a user was removed from group chat hidden group"
+        );
+        assert.equal(groupChat.audience.users[0]._id, data.users[0]._id,
+            "the correct user was removed from group chat hidden group"
+        );
+    }));
+
+    it("should not remove the user from private chat hidden groups", coroutine(function*() {
+        let chats = yield sessions[0]("GET", "/chats");
+        let privateChat = chats.find(chat => chat.isTwoPeople);
+        assert.equal(privateChat.audience.users.length, 2,
+            "no user was removed from private chat hidden group"
+        );
+    }));
+
+    it("should remove the user from folder hidden groups", coroutine(function*() {
+        let folders = yield sessions[0]("GET", "/folders");
+        let folder = folders.find(folder => !folder.defaultFolder);
+        assert.equal(folder.audience.users.length, 1,
+            "a user was removed from folder hidden group"
+        );
+        assert.equal(folder.audience.users[0], data.users[0]._id,
+            "the correct user was removed from folder hidden group"
+        );
+    }));
+
+    it("should not remove the user from Personal Files folder hidden group", coroutine(function*() {
+        let folders = yield sessions[0]("GET", "/folders");
+        let personalFolder = folders.find(folder =>
+            folder.defaultFolder && folder.name === "Personal Files");
+        assert.equal(personalFolder.audience.users.length, 1,
+            "no user was removed from Personal Files folder hidden group"
+        );
+    }));
 
```

Comment:
For at least one of these types of objects (event/chat/etc), there should be more than one created to make sure that the user is removed from all of them, not just one. This used to be a bug so a test should be added for it.